src/main/java/com/example/bookstoreweb/exception/CustomGlobalExceptionHandler.java

```diff
@@ -12,12 +13,28 @@
 import org.springframework.validation.ObjectError;
 import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.context.request.WebRequest;
 import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 
 @ControllerAdvice
 public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
 
+    @ExceptionHandler(RegistrationException.class)
+    public ResponseEntity<ErrorMessage> handleRegistrationException(RegistrationException e) {
+        return ResponseEntity
+                .status(HttpStatus.BAD_REQUEST)
```

Comment:
This status is more suitable for your usecase
```suggestion
                .status(HttpStatus.CONFLICT)
```